fix: notifications that dont mention a user explicitly

Poligoning opened the pull request:
fixed error: InvalidCharacterError: DOMTokenList.add: The token can not contain whitespace.
on: inbox/notifications

Poligoning committed:
fix: notifications

fixed error: InvalidCharacterError: DOMTokenList.add: The token can not contain whitespace.
on: inbox/notifications

katelyynn commented:
thank u for the pr !! in tha future pls dont change the code style as it doesnt match the rest of the codebase

this fix does work but i think a better fix is possible by checking what type of notification we're working with in the first place before checking a user's badges. as the issue with spaces in usernames comes from notifications that are about obsessions or listening reports (from what ive seen) that dont actually contain a username
![image](https://github.com/user-attachments/assets/0745c270-9de6-4c93-99ad-dee993c6e62f)

this should be possible by looking at the url and a simple `endsWith` + early return if its one of these two

- obsessions end with `/obsessions/set`
- listening reports end with `/listening-report/month` (you only get notifications per month iirc?)

you're free to take a shot in a new branch (or revert tha code style changes here) or i can work on this soonsies !! thank u still regardles :3

Poligoning committed:
fix: notifications

fixed error: InvalidCharacterError: DOMTokenList.add: The token can not contain whitespace.
on: inbox/notifications

Poligoning commented:
hopefully followed through this time around!! did it first thing when i woke up out of boredom so yeahh my apologies lol, awesome project though :)

katelyynn commented:
pretty much perfect now thank u :3 i cant seem to push to this branch (maybe because u didnt make a feature branch im not sure with github honestly..) so ill try request minor changes instead :3

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19830,27 +19830,24 @@ let has_prompted_for_update = false;
-                <h1>${trans[lang].inbox[page.subpage]}</h1>
-            </div>
-        `);
+       <div class="tag-side">
+           <div class="tag-icon ${(page.subpage == 'notifications') ? 'notifications' : 'inbox'}-icon"></div>
+       </div>
+       <div class="info-side">
+           <div class="sub-text">${trans[lang].inbox.name}</div>
+           <h1>${trans[lang].inbox[page.subpage]}</h1>
+       </div>
+   `);
```
think u could tab these back over pls:3

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19863,38 +19860,49 @@ let has_prompted_for_update = false;
+                let name_text = return_name_from_avatar(avatar_img);
+                let safeClassName = encodeURIComponent(name_text).replace(/%[0-9A-F]{2}/g, '_');
                 let badge = patch_avatar(avatar, name_text);
-                name.classList.add('notification-user-name', `user-status--bleh-${badge.type}`, `user-status--bleh-user-${name_text}`);
+
+                name.classList.add('notification-user-name',
+                    `user-status--bleh-${badge.type}`,
+                    `user-status--bleh-user-${safeClassName}`);
 
                 if (notification.classList.contains('inbox-notifications__item--highlight'))
-                    notification.classList.add('notification-user-name', `user-status--bleh-${badge.type}`, `user-status--bleh-user-${name_text}`);
+                    notification.classList.add('notification-user-name',
+                        `user-status--bleh-${badge.type}`,
+                        `user-status--bleh-user-${safeClassName}`);
```
`safeClassName` is probably not needed anymore as we're already checking if this is a user before attempting to grab badges. only reason there'd be spaces is if last.fm somehow allowed that at one point (which they might have done, horror)

if so u should be able to change it to
```js
let name_text = sanitise(bla bla bla);
```
and change safeClassName references back to name_text (sanitise is a bleh function)

katelyynn replied:
```suggestion
                    notification.classList.add('notification-user-name',
                        `user-status--bleh-${badge.type}`,
                        `user-status--bleh-user-${name_text}`);
```

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19863,38 +19860,49 @@ let has_prompted_for_update = false;
+                        `user-status--bleh-${badge.type}`,
+                        `user-status--bleh-user-${safeClassName}`);
             });
+
```
random gap :3

katelyynn replied:
```suggestion
```

katelyynn reviewed (changes requested):
should be good to go after this :>

Poligoning commented:
oh am i supposed to create new branches when proposing PRs, thought it worked differently first time even making one so my fault if i was!!

katelyynn commented:
> oh am i supposed to create new branches when proposing PRs, thought it worked differently first time even making one so my fault if i was!!

thats usually the way as it helps avoid potential conflicts (and also lets u propose multiple at once) but no worries literally everyone does it at first LMAO

katelyynn commented:
u should b able to apply the changes i think then its ready to merge

Poligoning commented:
> u should b able to apply the changes i think then its ready to merge

uhh how do i do this 😭, i dont see any buttons or anything unless im being silly

Poligoning commented:
all i see is this pretty much i dont think anything else is relevant

![image](https://github.com/user-attachments/assets/9ec165b9-775e-42bb-bba4-d3879232c17b)

(ive already like "resolved" the changes)

katelyynn commented:
umm if u un resolve tha comments there should b a commit suggestion button i think? i've not reviewed many pull requests so its new to me 😭 i would just push the changes myself but i dont have tha permissions on ur repo

katelyynn commented:
i can always apply the changes to bleh directly after merge if that's easier it's no hassle ^_^

Poligoning commented:
> umm if u un resolve tha comments there should b a commit suggestion button i think? i've not reviewed many pull requests so its new to me 😭 i would just push the changes myself but i dont have tha permissions on ur repo

uhh its just letting me agree to the changes/request a re-review, id need write access to the repo to do that im p sure? honestly completely lost its my first PR lol

Poligoning commented:
i mean thats the only thing that'd make sense to me as it literally tells me im good to go right above that

![image](https://github.com/user-attachments/assets/35f2f680-342d-401e-843d-cf26f9b5d9fe)

katelyynn commented:
![image](https://github.com/user-attachments/assets/7a918483-f2b9-476f-a30d-acba5f4e81db)

is ther any here o.O?

Poligoning commented:
> ![image](https://private-user-images.githubusercontent.com/46572320/407533779-7a918483-f2b9-476f-a30d-acba5f4e81db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMDU3MDIsIm5iZiI6MTczODEwNTQwMiwicGF0aCI6Ii80NjU3MjMyMC80MDc1MzM3NzktN2E5MTg0ODMtZjJiOS00NzZmLWEzMGQtYWNiYTVmNGU4MWRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDIzMDMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzZGUyNGEyMzJlYTdiYWI1ZTFmNjk0NDhiZDUyODkzMGZkYTFlMzBhNTI5ZDkxNTk5ZjIyY2Q5ODkxMWY4NTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tEgNQ3woWHXY9JXmXpkWSAOn8w_uRxULt6egrBP3t7s)
> 
> is ther any here o.O?

![image](https://github.com/user-attachments/assets/9f3681c7-be57-4b9c-8976-ef2a90a2bedb)

only being allowed to resolve it, if i mark it as resolved its just the opposite, actually let me give something a go real quick

* said something didn't work out 🙏, dont know how to do it if it can even be done tbh

Poligoning commented:
i mean its still telling me i need to look over the requested changes but i resolved all of them so im quite confused lol

katelyynn commented:
its no worries i think i submitted it a little wrong so it didnt give u an option :3 i can try again actually n see if not ill just do it after sorry for the trouble ^^

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19830,27 +19830,24 @@ let has_prompted_for_update = false;
-                <h1>${trans[lang].inbox[page.subpage]}</h1>
-            </div>
-        `);
+       <div class="tag-side">
+           <div class="tag-icon ${(page.subpage == 'notifications') ? 'notifications' : 'inbox'}-icon"></div>
+       </div>
+       <div class="info-side">
+           <div class="sub-text">${trans[lang].inbox.name}</div>
+           <h1>${trans[lang].inbox[page.subpage]}</h1>
+       </div>
+   `);
 
         page.structure.container.insertBefore(inbox_header, page.structure.container.firstElementChild);
```
```suggestion
            <div class="tag-side">
                <div class="tag-icon ${(page.subpage == 'notifications') ? 'notifications' : 'inbox'}-icon"></div>
            </div>
            <div class="info-side">
                <div class="sub-text">${trans[lang].inbox.name}</div>
                <h1>${trans[lang].inbox[page.subpage]}</h1>
            </div>
        `);

        page.structure.container.insertBefore(inbox_header, page.structure.container.firstElementChild);
```

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19863,38 +19860,49 @@ let has_prompted_for_update = false;
+                let avatar_img = avatar.querySelector('img');
+                if (!avatar_img) return;
 
+                let name_text = return_name_from_avatar(avatar_img);
+                let safeClassName = encodeURIComponent(name_text).replace(/%[0-9A-F]{2}/g, '_');
```
```suggestion
                let name_text = sanitise(return_name_from_avatar(avatar_img));
```

katelyynn commented on fm/bleh.user.js:
```diff
@@ -19863,38 +19860,49 @@ let has_prompted_for_update = false;
+
+                name.classList.add('notification-user-name',
+                    `user-status--bleh-${badge.type}`,
+                    `user-status--bleh-user-${safeClassName}`);
```
```suggestion
                    `user-status--bleh-user-${name_text}`);
```

katelyynn reviewed (changes requested):
ther we go sorry :3

Poligoning committed:
Update fm/bleh.user.js

Co-authored-by: katelyn <46572320+katelyynn@users.noreply.github.com>

Poligoning committed:
Update fm/bleh.user.js

Co-authored-by: katelyn <46572320+katelyynn@users.noreply.github.com>

Poligoning committed:
Update fm/bleh.user.js

Co-authored-by: katelyn <46572320+katelyynn@users.noreply.github.com>

Poligoning committed:
Update fm/bleh.user.js

Co-authored-by: katelyn <46572320+katelyynn@users.noreply.github.com>

Poligoning committed:
Update fm/bleh.user.js

Co-authored-by: katelyn <46572320+katelyynn@users.noreply.github.com>

Poligoning commented:
> its no worries i think i submitted it a little wrong so it didnt give u an option :3 i can try again actually n see if not ill just do it after sorry for the trouble ^^

err seems to still be the same weirdly, pretty sure it might just be a github moment now as its refusing to link me to the "requested changes" yet they're stopping me from merging lol... mb for the hassle and ty for the PR crash-course :)

Poligoning commented:
oh i just clicked submit rereview out of curiosity and its stuck on the resolved changes, i love git 😭

katelyynn reviewed (approved):
looks good ^^

katelyynn commented:
no worries about the PR stuff i think we both learned something LMAO but i can merge now i believe thank u <3

Poligoning commented:
> no worries about the PR stuff i think we both learned something LMAO but i can merge now i believe thank u <3

o yeah im not getting the error anymore? still cant merge myself but all green on my end now

Poligoning commented:
ty once more, awesome project :)

katelyynn commented:
^_^ will be in next minor release, ive moved it into my upcoming redesign branch too that splits into a separate file n credited u on the commit so its all good :3

katelyynn commented:
btw!! if u want a contributor badge on your profile in last.fm you can send me ur username in whichever way works (here, or u can dm my [kateshapedbox](https://www.last.fm/user/kateshapedbox) account)
![image](https://github.com/user-attachments/assets/2238c928-a0f4-4202-879d-f771e71f8d68)

Poligoning commented:
> btw!! if u want a contributor badge on your profile in last.fm you can send me ur username in whichever way works (here, or u can dm my [kateshapedbox](https://www.last.fm/user/kateshapedbox) account) ![image](https://private-user-images.githubusercontent.com/46572320/407598199-2238c928-a0f4-4202-879d-f771e71f8d68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMjUzMzIsIm5iZiI6MTczODEyNTAzMiwicGF0aCI6Ii80NjU3MjMyMC80MDc1OTgxOTktMjIzOGM5MjgtYTBmNC00MjAyLTg3OWQtZjc3MWU3MWY4ZDY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI5VDA0MzAzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjMDgxZjVmZjQ0YzFjNjhmM2UyYTJkY2YwYjdjYzBhYTM5MWVjNzM0NzM2ZWNjMTc3MmViNTI0Yzc5ZjU2MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nS0sNbV_-zrpIT9B3cry1WpBuTnCaBaP9JlQM5N9GEI)

ohhh that would be hella cool, https://www.last.fm/user/Poligoner, ty!!
